packages/calling-stateful-client/src/CallAgentDeclarativeCommon.ts

```diff
@@ -123,8 +130,16 @@ export abstract class ProxyCallAgentCommon {
         new IncomingCallSubscriber(incomingCall, this.setIncomingCallEnded)
       );
     }
-    this._declarativeIncomingCalls.set(incomingCall.id, incomingCallDeclaratify(incomingCall, this._context));
-    this._context.setIncomingCall(convertSdkIncomingCallToDeclarativeIncomingCall(incomingCall));
+    if (incomingCall.kind === 'TeamsIncomingCall') {
```

Comment:
Let's make it a type guard instead of using force cast, just in case we need to use it in the future